webpack/components/AnsibleRolesAndVariables/AnsibleRolesAndVariables.js

```diff
@@ -0,0 +1,223 @@
+        params: { changed: params, proxy },
+        handleSuccess: () => dispatch(push(ANSIBLE_ROLES_INDEX)),
+      })
+    );
+  };
+
+  const onCancel = () => {
+    dispatch(push(ANSIBLE_ROLES_INDEX));
+  };
+
+  const handleSetPage = (event, newPage, perPage, startIdx, endIdx) =>
+  {
+    setPage(newPage);
+    setPaginatedRows(rows.slice(startIdx, endIdx))
+  };
+
+  const handlePerPageSelect = (event, newPerPage, newPage, startIdx, endIdx) =>
+  {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+  };
+
+  const renderPagination = (variant = 'top') =>
+  {
+    if (rows.length !== 0)
+    {
+      return (
+        <Pagination
+          isCompact
+          itemCount={rows.length}
+          page={page}
+          perPage={perPage}
+          defaultToFullPage
+          onSetPage={handleSetPage}
+          onPerPageSelect={handlePerPageSelect}
+          perPageOptions={[
+            { title: '3', value: 3 },
+            { title: '5', value: 5 },
+            { title: '10', value: 10 },
+          ]}
+          titles={{
+            paginationTitle: `${variant} pagination`
+          }}
+        />
+      );
+    };
+  };
+
+  const renderSelectAll = () =>
+  {
+    if (rows.length !== 0)
+    {
+      return(
+        <Toolbar>
+          <ToolbarContent>
+            <ToolbarGroup variant="icon-button-group">
+              <ToolbarItem>
+              <Checkbox
+                isChecked={isChecked}
+                onChange={SelectAll}
+                aria-label="select all checkbox"
+                id="select-all"
+                name="select-all"
+                label={isChecked ? 'Deselect all' : 'Select all'}
+               />
+              </ToolbarItem>
+            </ToolbarGroup>
+          </ToolbarContent>
+        </Toolbar>
+      );
+    };
+  };
+
+  const renderNotFound = () =>
+  {
+    if (rows.length === 0)
+    {
+      return(
+        <EmptyState>
+          <EmptyStateIcon icon={SearchIcon} />
+          <Title headingLevel="h2" size="lg"> No Roles found </Title>
+          <EmptyStateBody> All Roles and variables are synced are imported from the smart proxy</EmptyStateBody>
+        </EmptyState>
+      );
+    };
+  };
+
+  const renderSubmitAndCancel = () =>
+  {
+    if (rows.length !==0 )
+    {
+      return(
+        <div>
+          <br />
+          <br />
+          <Button variant="primary" onClick={onSubmit}> Submit </Button>
+          <Button variant="secondary" onClick={onCancel}> Cancel </Button>
+        </div>
+      );
+    };
+  };
+
+  return (
+    <React.Fragment>
+      {renderSelectAll()}
+      <Table
+        aria-label="import roles and variables"
+        onSelect={onSelect}
+        rows={paginatedRows}
+        cells={columns}
+        canSelectAll={false}
+      >
+        <TableHeader />
+        <TableBody />
+      </Table>
+      {renderNotFound()}
+      {renderPagination()}
+      {renderSubmitAndCancel()}
```

Comment:
You can turn these into separate components to make this component a bit more slim.